packages/test/test-end-to-end-tests/src/test/summarization/lostSnapshot.spec.ts

```diff
@@ -0,0 +1,119 @@
+ */
+
+import { strict as assert } from "assert";
+
+import { ITestDataObject, describeCompat, itExpects } from "@fluid-private/test-version-utils";
+import type { ContainerRuntime } from "@fluidframework/container-runtime/internal";
+import type { IDocumentStorageService } from "@fluidframework/driver-definitions/internal";
+import {
+	ITestContainerConfig,
+	ITestObjectProvider,
+	createSummarizer,
+	createTestConfigProvider,
+	summarizeNow,
+} from "@fluidframework/test-utils/internal";
+
+/**
+ * These tests validate the behavior of the summarizer when it gets summary acks that are newer than the summary
+ * it knows about.
+ */
+describeCompat("Summarization - lost snapshot", "NoCompat", (getTestObjectProvider) => {
+	const testContainerConfig: ITestContainerConfig = {
+		runtimeOptions: {
+			summaryOptions: {
+				summaryConfigOverrides: { state: "disabled" },
+			},
+		},
+	};
+	const configProvider = createTestConfigProvider();
+
+	let provider: ITestObjectProvider;
+
+	beforeEach("getTestObjectProvider", async function () {
+		provider = getTestObjectProvider({ syncSummarizer: true });
+		if (provider.driver.type !== "local") {
+			this.skip();
+		}
+		configProvider.set("Fluid.ContainerRuntime.Test.CloseSummarizerDelayOverrideMs", 0);
+	});
+
+	afterEach(() => {
+		configProvider.clear();
+	});
+
+	interface IHackableCollection {
+		deleteOne(query: { _id: string }): void;
+	}
+
+	interface IHackableHistorian {
+		readonly trees: IHackableCollection;
+		readonly commits: IHackableCollection;
+		updateRef(ref: string, params: { sha: string }): void;
+	}
+
+	interface IHackableGitManager {
+		readonly historian: IHackableHistorian;
+	}
+
+	interface IHackLocalDocumentStorageService extends IDocumentStorageService {
+		readonly manager: IHackableGitManager;
+	}
+
+	itExpects(
+		"lost a snapshot but summary ack and op are still in op stream",
+		[{ eventName: "fluid:telemetry:Summarizer:Running:SummarizeFailed" }],
+		async () => {
+			const container1 = await provider.makeTestContainer(testContainerConfig);
+			const defaultDataStore = (await container1.getEntryPoint()) as ITestDataObject;
+			defaultDataStore._root.set("1", "2");
+
+			const summarizer1 = await createSummarizer(provider, container1, {
+				loaderProps: { configProvider },
+			});
+
+			const containerRuntime1 = defaultDataStore._context.containerRuntime as ContainerRuntime;
+			const localDocumentStorageService = (containerRuntime1.storage as any)._storageService
+				.internalStorageService
+				.internalStorageService as unknown as IHackLocalDocumentStorageService;
+
+			// Get original snapshot
+			const versions = await localDocumentStorageService.getVersions(null, 10);
+			const originalVersion = versions[0];
+
+			const { summaryVersion } = await summarizeNow(summarizer1.summarizer);
+			await provider.ensureSynchronized();
+
+			// Delete server snapshot (local server specific implementation)
+			const gitManager = localDocumentStorageService.manager;
```

Comment:
This test is very hacky and likely candidate to be a flaky / failing test in the future. It would be fine if we didn't have any other way to repro but I did repro it in a very simple way. See https://github.com/microsoft/FluidFramework/pull/21761.

Reply:
Ok, I'll close it